src/main/java/it/uniroma3/mat/extendedset/intset/ImmutableConciseSet.java

```diff
@@ -694,19 +687,26 @@ public static ImmutableConciseSet doIntersection(Iterator<ImmutableConciseSet> s
     return new ImmutableConciseSet(IntBuffer.wrap(retVal.toArray()));
   }
 
-  private static void sort(final WordIterator[] a, final int to, final Comparator<WordIterator> comp)
+  /**
+   * Variation of insertion sort, elements [maxChangedIndex + 1, size) are sorted, elements [0, maxChangedIndex] should
+   * be inserted into that sorted range.
+   */
+  private static void partialSort(
```

Comment:
`Theoretical limit is the number of dimensions in data sources.`

Regex filters make ephemeral bitsets, so I don't think that's the limit. It could be up to the cardinality of dimension values in some weird cases. Its been a long time since I looked at regex stuff though.